src/Raven.Client/Documents/Session/EntityToBlittable.cs

```diff
@@ -283,7 +285,26 @@ private static bool TryRemoveIdentityProperty(BlittableJsonReaderObject document
         {
             var identityProperty = conventions.GetIdentityProperty(entityType);
             if (identityProperty == null)
+            {
+                if (conventions.AddIdFieldToDynamicObjects && entityType.GetInterfaces().Contains(typeof(IDynamicMetaObjectProvider)))
```

Comment:
We need to cache this, it is too expensive. Probably better to pass the entity from the calling method and just to `entity is IDynamicMetaObjectProvider`

Reply:
Yes, passing it there is much better

Reply:
done